Account/FxACommandsClient.swift

```diff
@@ -0,0 +1,305 @@
+        return account.marriedState() >>== { marriedState in
+            let sessionToken = marriedState.sessionToken as NSData
+            let client = FxAClient10(authEndpoint: self.account.configuration.authEndpointURL)
+            client.devices(withSessionToken: sessionToken) >>== { response in
+                let devices = response.devices
+
+                for command in commands {
+                    guard let commandName = command.data["command"].string,
+                        let payload = command.data["payload"].string,
+                        let senderDeviceID = command.data["sender"].string,
+                        let sender = devices.find({ $0.id == senderDeviceID }) else {
+                            continue
+                    }
+
+                    switch commandName {
+                    case FxACommandSendTab.Name:
+                        self.sendTab.handle(sender: sender, encrypted: payload)
+                    default:
+                        log.info("Unknown command: \(commandName)")
+                    }
+                }
+            }
+        }
+    }
+}
+
+open class FxACommandSendTabKeys: JSONLiteralConvertible {
+    open var publicKey: String
+    open var privateKey: String
+    open var authSecret: String
+
+    public init(publicKey: String, privateKey: String, authSecret: String) {
+        self.publicKey = publicKey
+        self.privateKey = privateKey
+        self.authSecret = authSecret
+    }
+
+    open func asJSON() -> JSON {
+        return JSON([
+            "publicKey": self.publicKey,
+            "privateKey": self.privateKey,
+            "authSecret": self.authSecret
+        ])
+    }
+}
+
+open class FxACommandSendTab {
+    public static let Name = "https://identity.mozilla.com/cmd/open-uri"
+
+    let commandsClient: FxACommandsClient
+    let account: FirefoxAccount
+    let sendTabKeysCache: KeychainCache<FxACommandSendTabKeys>
+
+    init(commandsClient: FxACommandsClient, account: FirefoxAccount) {
+        self.commandsClient = commandsClient
+        self.account = account
+        self.sendTabKeysCache = KeychainCache(branch: "account.sendTabKeys", label: account.stateCache.label, value: nil)
+    }
+
+    public func send(to devices: [FxADevice], url: String, title: String) {
+        let json = JSON([
+            "entries": [["title": title, "url": url]]
+        ])
+
+        guard let jsonString = json.stringValue() else {
+            return
+        }
+
+        for device in devices {
+            encrypt(message: jsonString, device: device) >>== { encryptedPayload in
+                self.commandsClient.invoke(commandName: FxACommandSendTab.Name, toDevice: device, withPayload: encryptedPayload).bind { result in
+                    return deferMaybe(result.isSuccess)
+                }
+            }
+        }
+
+        // TODO: gather stats here about success/failures
+    }
+
+    public func isDeviceCompatible(_ device: FxADevice) -> Bool {
```

Comment:
I would normally agree, but I tried to keep the naming conventions as close as possible to what's in the desktop implementation so that it is easier to reference/check things.

Reply:
For all these "just keeping the same names as the desktop implementation", perhaps we could wrap those choices up in a protocol?